### PR TITLE
Tests refactor

### PR DESCRIPTION
**Pull request by DIOHz0r**
Signed-off-by: Domingo Oropeza <doropeza@teclib.com>

### Changes description

Small refactor of code for tests.

### Checklist

Please check if your PR fulfills the following specifications:

- [ ] Tests for the changes have been added
- [ ] Docs have been added/updated

### Estimated time

|Assignee|:tomato:|
|:---|:---:|
|@DIOHZ0r |9|

### References

Closes #N/A
Related #N/A
Depends on #N/A

### CODE REVIEW

File: tests/suite-integration/PluginFlyvemdmMqttuser.php [hunk 1]
```diff
@@ -38,15 +38,15 @@ class PluginFlyvemdmMqttuser extends CommonTestCase {
    /**
     * @return array
     */
-   protected function mqttuserProvider() {
+   protected function providerMqttuser() {
```
Comment (btry) [inline tests/suite-integration/PluginFlyvemdmMqttuser.php]: the provider should be ```providerAddAndDeleteMqttuser```

File: tests/suite-unit/PluginFlyvemdmCommon.php [hunk 2]
```diff
@@ -71,7 +71,7 @@ public function testGenerateUUID() {
    /**
     * @return array
     */
-   protected function resourcesProvider() {
+   protected function providerResources() {
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmCommon.php]: the provider should be ```providerRecursiveRmdir```

File: tests/suite-unit/PluginFlyvemdmCommon.php [hunk 3]
```diff
@@ -137,7 +137,7 @@ public function testGetMax() {
     * Used as provider for start/ends with methods
     * @return array
     */
-   protected function stringProvider() {
+   protected function providerString() {
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmCommon.php]: the provider should be ```providerStartsWith```

File: tests/suite-unit/PluginFlyvemdmCommon.php [hunk 4]
```diff
@@ -168,7 +168,7 @@ public function testStartsWith($data, $expected) {
 
    /**
     * @tags testEndsWith
-    * @dataProvider stringProvider
+    * @dataProvider providerString
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmCommon.php]: create a method ```providerEndsWith``` which returns the result from ```providerStartsWith```

File: tests/suite-unit/PluginFlyvemdmFleet.php [hunk 5]
```diff
@@ -115,7 +118,7 @@ public function testShowForm() {
          ->contains('input type="hidden" name="_glpi_csrf_token"');
    }
 
-   protected function inputAddProvider() {
+   protected function providerInputAdd() {
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmFleet.php]: the provider should be ```providerPrepareInputForAdd```

File: tests/suite-unit/PluginFlyvemdmPolicyDeployapplication.php [hunk 6]
```diff
@@ -52,7 +52,7 @@ private function createNewPolicyInstance() {
       return [$policy, $policyData];
    }
 
-   protected function validationProvider() {
+   protected function providerValidation() {
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmPolicyDeployapplication.php]: the provider should be ```providerCreatePolicy```

File: tests/suite-unit/PluginFlyvemdmPolicyDeployfile.php [hunk 7]
```diff
@@ -42,7 +42,7 @@ class PluginFlyvemdmPolicyDeployfile extends CommonTestCase {
       'unicity'   => '0',
    ];
 
-   protected function validationProvider() {
+   protected function providerValidation() {
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmPolicyDeployfile.php]: the provider should be ```providerCreatePolicy```

File: tests/suite-unit/PluginFlyvemdmPolicyRemoveapplication.php [hunk 8]
```diff
@@ -58,7 +58,7 @@ private function createNewPolicyInstance($policyData = null) {
    /**
     * @return array
     */
-   protected function policyInputProvider() {
+   protected function providerPolicyInput() {
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmPolicyRemoveapplication.php]: the provider should be ```providerIntegrityCheck```

File: tests/suite-unit/PluginFlyvemdmPolicyRemoveapplication.php [hunk 9]
```diff
@@ -135,7 +135,7 @@ public function testUnicityCheck(array $data, array $expected) {
    /**
     * @return array
     */
-   public function applyProvider() {
+   public function providerApply() {
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmPolicyRemoveapplication.php]: it seems this provider is not used

File: tests/suite-unit/PluginFlyvemdmCommon.php [hunk 10]
```diff
@@ -137,7 +137,7 @@ public function testGetMax() {
     * Used as provider for start/ends with methods
     * @return array
     */
-   protected function stringProvider() {
+   protected function providerStartsAndEndWith() {
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmCommon.php]: it is more generic to have an other provider for testEndsWith. The goal is to force a matching between a provider and a test.

```php 
public function providderEndsWith() {
   return $this->providerStartsWith();
}
```

File: tests/suite-unit/PluginFlyvemdmCommon.php [hunk 11]
```diff
@@ -156,7 +156,7 @@ protected function stringProvider() {
 
    /**
     * @tags testStartsWith
-    * @dataProvider stringProvider
+    * @dataProvider providerStartsAndEndWith
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmCommon.php]: see above comment

File: tests/suite-unit/PluginFlyvemdmPolicyDeployfile.php [hunk 12]
```diff
@@ -247,9 +247,9 @@ public function testConflictCheck() {
    }
 
    /**
-    * @tags testUnApply
+    * @tags testPreUnapply
     */
-   public function testPre_unapply() {
+   public function testPreUnapply() {
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmPolicyDeployfile.php]: I disagree: the tested method is ```pre_unapply``` with underscore.

File: tests/suite-unit/PluginFlyvemdmCommon.php (hunk 10 above)
Comment (DIOHz0r) [reply tests/suite-unit/PluginFlyvemdmCommon.php]: That's why it was only providerString, because it is used for both test

Comment (btry) [reply tests/suite-unit/PluginFlyvemdmCommon.php]: but we loose the relationship between name of provider and name of test

**Pull request merged**